### PR TITLE
Color module enhancements

### PR DESCRIPTION
**Pull request by AntonBallmaier**
## Changelog / Overview

- Replaced `BLUE_E` with what was previously `DARK_BLUE` and removed `DARK_BLUE`
- Added alias `LIGHTER_GRAY` for `GRAY_A`
- Added `PURE_RED`, `PURE_BLUE` and renamed `GREEN_SCREEN` to `PURE_GREEN`
- All gray colors are now also available using British spelling (including `GREY_BROWN`)
- Replaced color example in the docs. It can now be used as a quick reference for all color names.

## Motivation
After seeing /u/restinworld's [color wheel video](https://www.youtube.com/watch?v=n9uUunuCcLk) I wanted to improve the colors module...

## Explanation for Changes
In addition to the changes in the changelog I sorted the colors in the enum and made name aliases reference the first name of that color instead of redefining the same value. Furthermore I added type hints to the conversion functions.

Docs reference: https://manimce--1504.org.readthedocs.build/en/1500/reference/manim.utils.color.Colors.html

## Further Comments
I am unsure if I should name the colors in the example in uppercase...

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by kolibril13** [comment]
Amazing pr!
Two thoughts (only optional, but feel free to add it :) ).
There could be a note to say:
:: note
   The color of type "C" is equal to the colornames without a letter, e.g. GREEN = GREEN_C

And about this blog:
image
Is `import manim.utils.color as C` really the way that we want to use colors?
I would not mention it there. But only a personal optionion.

**Review by kolibril13 — APPROVED** [review]

**Comment by kolibril13** [comment]
oh, before merging: why did the test data change?

**Comment by AntonBallmaier** [comment]
> Is `import manim.utils.color as C` really the way that we want to use colors?

No, probably not. I'll change that.

**Comment by AntonBallmaier** [comment]
> The color of type "C" is equal to the colornames without a letter, e.g. GREEN = GREEN_C

Ill add that

**Comment by AntonBallmaier** [comment]
I also forgot to include the new pure colors in the example.

File: manim/utils/color.py [hunk 1]
```diff
@@ -294,46 +350,45 @@ def color_to_rgb(color):
...
-    try:
-        return Color(rgb=rgb)
-    except Exception:
-        return Color(WHITE)
+def rgb_to_color(rgb: Iterable[float]) -> Color:
+    return Color(rgb=rgb)
 
 
-def rgba_to_color(rgba):
+def rgba_to_color(rgba: Iterable[float]) -> Color:
```
Comment (kolibril13) [inline manim/utils/color.py:361]: Why Iterable and not sequence?

**Comment by kolibril13** [comment]
nice!
And what about these new test data?
image

**Comment by AntonBallmaier** [comment]
> And what about these new test data?
Since I removed dark_blue and replaced blue_e with it's value, the scenes using blue_e changed a tiny bit.

File: manim/utils/color.py (hunk 1 above)
Comment (AntonBallmaier) [reply manim/utils/color.py:361]: I because I thought it was specific enough. Is it not?

Comment (kolibril13) [reply manim/utils/color.py:381]: I remember that I looked up once that Sequence is a bit more specfic then Iterable, but I can not find the link anymore. As it does not make a huge difference, we can stay with Iterable here :-)

**Comment by kolibril13** [comment]
LGTM now!

**Pull request merged**